components/scim/org.wso2.carbon.identity.scim.provider/src/main/java/org/wso2/carbon/identity/scim/provider/impl/SCIMUserManager.java

```diff
@@ -1059,6 +1075,23 @@ public Group patchGroup(Group oldGroup, Group newGroup) throws CharonException {
                 List<String> deleteRequestedMembers =
                         newGroup.getMembersWithDisplayName(SCIMConstants.CommonSchemaConstants.OPERATION_DELETE);
 
+                int noOfAddedMembers = addRequestedMembers.size();
+                int noOfDeletedMembers = deleteRequestedMembers.size();
+
+                for (int i = 0; i < noOfAddedMembers; i++) {
+                    String addRequestMemberUpperCaseDomain = UserCoreUtil
+                            .addDomainToName(UserCoreUtil.removeDomainFromName(addRequestedMembers.get(i)),
+                                    UserCoreUtil.extractDomainFromName(addRequestedMembers.get(i)).toUpperCase());
+                    addRequestedMembers.add(addRequestMemberUpperCaseDomain);
+                }
+
+                for (int i = 0; i < noOfDeletedMembers; i++) {
+                    String deleteRequestMemberUpperCaseDomain = UserCoreUtil
+                            .addDomainToName(UserCoreUtil.removeDomainFromName(deleteRequestedMembers.get(i)),
+                                    UserCoreUtil.extractDomainFromName(deleteRequestedMembers.get(i)).toUpperCase());
+                    deleteRequestedMembers.add(deleteRequestMemberUpperCaseDomain);
```

Comment:
You seem to be handling the new array list size properly but its confusing for a new developer. You could use List.add(index, element) method so you can override the existing elements instead of adding it as a new element at the end.